### ISSUE TITLE
[cocoon] Throttle swarming presubmit tasks during peak hours of users consuming a large amount of infra quota

### ISSUE THREAD

**Issue by CaseyHillers** [flutter/flutter#138511]
Occasionally, someone will send a large number of cleanup PRs which consumes a large chunk of presubmit capacity. To protect overall SLOs for the community, we should deprioritize this user for the day.

Peak hours would be around 10 AM - 3 PM PST on US workdays.

This could be some metric based on the number of builds being scheduled. For example, we could batch it in groups of 50 targets. These numbers can be tweaked to better match our infra capacity:

0 - 150: No impact
150 - 250: Slight reduction
250+: Lowest priority

Cocoon can leave a comment on the PR as an FYI that the user is being throttled for using too much capacity.

**Comment by ricardoamador** [comment]
The primary consumers of quota for presubmit are the roller PRs right now. Even in post submit. It's not a user consuming the quota.

![image](https://github.com/flutter/flutter/assets/32242716/59d4dcf1-f051-42e9-82ae-f07c59f28da6)

The rollers tie up resources hourly and are a huge capacity hit. Many rollers an hour sometimes on top of the other work being done. Why not throttle those?

**Comment by matanlurey** [comment]
I agree with Ricardo, I have not (personally) seen any (human) user write 100+ PRs a day.

**Comment by CaseyHillers** [comment]
Ideally, we'd throttled against CPU used, but I'm not sure it's easy to pull that data from swarming. So I suggest targets as it's something the scheduler is already aware of. For framework commits, this would incur a cost of 20~70 targets versus packages being ~10 targets.

We'd have a special group for users that we expect to use a large amount of quota (like the engine-flutter-autoroll bot). They can still get throttled if their increase frequency or increase number of targets run, and we can always revisit those numbers if needed.

For concerns around bot capacity use, that's still an active area of work. For example, last week I optimized Fuchsia usage in https://github.com/flutter/flutter/issues/138087 as it was sending 4 engine commits a day, which could result in 4 framework commits a day. Now it's at 1 engine commit a day, resulting in 1 framework commit a day. There seems to be a few other areas we can work on (such as the flakiness bot and dependabot).

**Comment by matanlurey** [comment]
Nice. Let me know if I can help at all from an engine POV, otherwise happy throttling!
